examples/bri-3/front-end/src/login/Login.js

```diff
@@ -1,38 +1,65 @@
-  if (window.ethereum) {
-    const accounts = await window.ethereum.request({ method: 'eth_requestAccounts' });
-    const nonceDto = {
-      publicKey: accounts[0]
-    };
-    const nonce = await axios.post(`${process.env.REACT_APP_SRI_BACKEND}/auth/nonce`, nonceDto);
-    const signature = await window.ethereum.request({
-      method: 'personal_sign',
-      params: [nonce.data, accounts[0]],
-    });
 
-    const loginDto = {
-      message: nonce.data,
-      signature,
-      publicKey: accounts[0],
-    };
+export default function Login() {
+  const [data, setData] = useState({
+    publicKey: '',
+    token: ''
+  });
+
+  const handleLogin = async () => {
+    try {
+    // TODO: handle situations like account switch in metamask or when wallet is initially locked
+    if (window.ethereum) {
+      const accounts = await window.ethereum.request({ method: 'eth_requestAccounts' });
+      const nonceDto = {
+        publicKey: accounts[0]
+      };
+      const nonce = await axios.post(`${process.env.REACT_APP_SRI_BACKEND}/auth/nonce`, nonceDto);
+      const signature = await window.ethereum.request({
+        method: 'personal_sign',
+        params: [nonce.data, accounts[0]],
+      });
+
+      const loginDto = {
+        message: nonce.data,
+        signature,
+        publicKey: accounts[0],
+      };
+
+      const tokenRequest = await axios.post(`${process.env.REACT_APP_SRI_BACKEND}/auth/login`, loginDto);
 
-    const tokenRequest = await axios.post(`${process.env.REACT_APP_SRI_BACKEND}/auth/login`, loginDto);
+      if (tokenRequest?.data?.access_token) {
+        const publicKey = accounts[0];
+        const token = tokenRequest.data.access_token;
+        setData({ publicKey, token });
+        toast.success('Success!');
 
-    // TODO: add toast library and show success/failure login with error messages
-    // TODO: add logged in user details on UI
-    if (tokenRequest?.data?.access_token) {
-      console.log(tokenRequest.data.access_token);
+      } else {
+        toast.error('No token.');
+      }
     } else {
-      console.log('No token');
+      throw new Error('No MM found. Please install Metamask');
     }
-  } else {
-    throw new Error('No MM found. Please install Metamask');
-  }
-};
-
-export default function Login() {
-  return <button  data-testid="login" onClick={handleLogin}>Login</button>;
+  } catch(error) {
+    toast.error(error.message);
+  };
+}
+  
+  return (
+    <>
+      <ToastContainer />
+      <p>Welcome back 🎉 !</p> <br />
+      <Button data-testid="login" onClick={handleLogin}>Login</Button> <br />
+      <Card style={{textAlign:'left', width: '50rem', fontSize: 16, color: 'white', backgroundColor: 'transparent', border: 'transparent', padding: '10px'}}> 
+        {data.publicKey && <Card.Text><b>Current Public Key: </b>{data.publicKey}</Card.Text>}
+        {data.token && <Card.Text><b>Your Access Token: </b>{data.token}</Card.Text>}
```

Comment:
is there a reason to display access token?

Reply:
It's just from my speculation when I thinking about any info related to user info. I can delete it!

Reply:
I would ask not to display this information. @Therecanbeonlyone1969 what are your views on this?

Reply:
maybe it would make more sense to fetch bpi subject info after login and show that, i dont think showing access token is useful

Reply:
Yep, just remove the access token and its good.

Reply:
@mkxx712  Please remove access token and request approval again

Reply:
Thanks for all the comments! I'll fix them